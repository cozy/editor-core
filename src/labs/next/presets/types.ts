import { FeatureFlags } from '../../../plugins/feature-flags-context/types';

export interface EditorPresetProps {
  excludes?: Set<string>;
  experimental?: Array<string>;
  featureFlags?: FeatureFlags;
}
