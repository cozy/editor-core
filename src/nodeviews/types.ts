import type { Node as PMNode } from '@atlaskit/editor-prosemirror/model';

export interface ReactNodeProps {
  selected: boolean;
}

export type ProsemirrorGetPosHandler = () => number | undefined;
export type getPosHandler = getPosHandlerNode | boolean;
export type getPosHandlerNode = () => number | undefined;
export type ReactComponentProps = { [key: string]: unknown };
export type ForwardRef = (node: HTMLElement | null) => void;
export type shouldUpdate = (nextNode: PMNode) => boolean;
