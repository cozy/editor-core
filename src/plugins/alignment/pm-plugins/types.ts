export type AlignmentState = 'start' | 'end' | 'center';
export type AlignmentPluginState = {
  align: AlignmentState;
  isEnabled?: boolean;
};
