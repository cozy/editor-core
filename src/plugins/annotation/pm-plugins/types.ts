import type { DecorationSet } from '@atlaskit/editor-prosemirror/view';
import type {
  EditorState,
  SelectionBookmark,
} from '@atlaskit/editor-prosemirror/state';
import type { PortalProviderAPI } from '@atlaskit/editor-common/portal-provider';
import type { EditorAnalyticsAPI } from '@atlaskit/editor-common/analytics';
import type {
  Dispatch,
  EventDispatcher,
} from '@atlaskit/editor-common/event-dispatcher';
import type { InlineCommentAnnotationProvider, AnnotationInfo } from '../types';

export enum ACTIONS {
  UPDATE_INLINE_COMMENT_STATE,
  SET_INLINE_COMMENT_DRAFT_STATE,
  INLINE_COMMENT_UPDATE_MOUSE_STATE,
  INLINE_COMMENT_CLEAR_DIRTY_MARK,
  ADD_INLINE_COMMENT,
  INLINE_COMMENT_SET_VISIBLE,
  CLOSE_COMPONENT,
  SET_SELECTED_ANNOTATION,
}

export interface InlineCommentPluginOptions {
  dispatch: Dispatch;
  eventDispatcher: EventDispatcher;
  portalProviderAPI: PortalProviderAPI;
  provider: InlineCommentAnnotationProvider;
  editorAnalyticsAPI: EditorAnalyticsAPI | undefined;
}
export interface InlineCommentMouseData {
  isSelecting: boolean;
}

export type InlineCommentMap = { [key: string]: boolean };
export type InlineCommentAction =
  | {
      type: ACTIONS.UPDATE_INLINE_COMMENT_STATE;
      data: InlineCommentMap;
    }
  | {
      type: ACTIONS.SET_INLINE_COMMENT_DRAFT_STATE;
      data: {
        drafting: boolean;
        editorState: EditorState;
      };
    }
  | {
      type: ACTIONS.INLINE_COMMENT_UPDATE_MOUSE_STATE;
      data: {
        mouseData: InlineCommentMouseData;
      };
    }
  | { type: ACTIONS.INLINE_COMMENT_CLEAR_DIRTY_MARK }
  | { type: ACTIONS.CLOSE_COMPONENT }
  | {
      type: ACTIONS.ADD_INLINE_COMMENT;
      data: {
        drafting: boolean;
        inlineComments: InlineCommentMap;
        editorState: EditorState;
        selectedAnnotations: AnnotationInfo[];
      };
    }
  | { type: ACTIONS.INLINE_COMMENT_SET_VISIBLE; data: { isVisible: boolean } }
  | {
      type: ACTIONS.SET_SELECTED_ANNOTATION;
      data: { selectedAnnotations: AnnotationInfo[] };
    };

export type InlineCommentPluginState = {
  annotations: InlineCommentMap;
  selectedAnnotations: AnnotationInfo[];
  dirtyAnnotations?: boolean;
  mouseData: InlineCommentMouseData;
  draftDecorationSet?: DecorationSet;
  bookmark?: SelectionBookmark;

  // Denotes if annotations are allowed to be create on empty nodes or nodes of whitespace (Confluence spec)
  disallowOnWhitespace: boolean;

  // Allow users to hide inline comments during editing
  isVisible: boolean;
  skipSelectionHandling: boolean;
};
