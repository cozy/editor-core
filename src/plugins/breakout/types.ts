import type { ContentNodeWithPos } from '@atlaskit/editor-prosemirror/utils';
export interface BreakoutPluginState {
  breakoutNode: ContentNodeWithPos | undefined;
}
