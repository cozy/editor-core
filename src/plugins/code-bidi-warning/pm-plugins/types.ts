import type { DecorationSet } from '@atlaskit/editor-prosemirror/view';

export type CodeBidiWarningPluginState = {
  decorationSet: DecorationSet;
  codeBidiWarningLabel: string;
  tooltipEnabled: boolean;
};
