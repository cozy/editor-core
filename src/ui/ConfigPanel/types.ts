export type FormResult = {
  [key: string]: string | number | string[] | number[] | undefined;
};

export enum ValidationError {
  Required = 'required',
  Invalid = 'invalid',
}

export enum FieldTypeError {
  isMultipleAndRadio = 'isMultipleAndRadio',
}

export type Entry<T> = [string, T];
export type OnBlur = (name: string) => void;
