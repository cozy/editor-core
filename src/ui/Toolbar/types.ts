import React from 'react';
import { UiComponentFactoryParams } from '../../types/ui-components';

export enum ToolbarSize {
  XXL = 6,
  XL = 5,
  L = 4,
  M = 3,
  S = 2,
  XXXS = 1,
}

export type ToolbarUiComponentFactoryParams = UiComponentFactoryParams & {
  toolbarSize: ToolbarSize;
  isToolbarReducedSpacing: boolean;
  isLastItem?: boolean;
};
export type ToolbarUIComponentFactory = (
  params: ToolbarUiComponentFactoryParams,
) => React.ReactElement<any> | null;
