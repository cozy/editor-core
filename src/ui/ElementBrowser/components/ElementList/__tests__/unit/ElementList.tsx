const mockGetWidth = jest.fn();

jest.mock('../../../../hooks/use-container-width', () => {
  const originalHook = jest.requireActual(
    '../../../../hooks/use-container-width',
  ).default;
  return (...args: any) => ({
    ...originalHook(...args),
    containerWidth: mockGetWidth(),
  });
});

import React from 'react';
import { ReactWrapper, mount } from 'enzyme';
import { ELEMENT_LIST_PADDING } from '../../../../constants';
import ElementList from '../../ElementList';
import { Modes } from '../../../../types';

const props = {
  items: [
    { name: 'item-1', title: 'Item 1', action: jest.fn() },
    { name: 'item-2', title: 'Item 2', action: jest.fn() },
    { name: 'item-3', title: 'Item 3', action: jest.fn() },
    { name: 'item-4', title: 'Item 4', action: jest.fn() },
    { name: 'item-5', title: 'Item 5', action: jest.fn() },
  ],
  mode: Modes.full,
  onInsertItem: jest.fn(),
  onEnterKeyPress: jest.fn(),
  selectedItemIndex: 1,
  setSelectedItemIndex: jest.fn(),
  setFocusedItemIndex: jest.fn(),
  setColumnCount: jest.fn(),
};

let wrapper: ReactWrapper;

describe('ElementList', () => {
  it('should match snapshot', () => {
    const wrapper = mount(<ElementList {...props} />);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
  describe('Collection', () => {
    it('renders a Virtualized Collection component with the right width', () => {
      mockGetWidth.mockReturnValueOnce(660);
      wrapper = mount(<ElementList {...props} />);
      const Collection = wrapper.find('Collection');
      expect(Collection).toHaveLength(1);
      expect(Collection.props().width).toStrictEqual(
        660 - ELEMENT_LIST_PADDING * 2,
      );
    });
    /**
     * Needs validation as there's a hack to remove tabIndex from react-virtualized/Collection.
     * If this className ever changes, the functionality may break.
     * ClassNames are specified here:
     * https://github.com/bvaughn/react-virtualized/blob/master/docs/Collection.md#class-names
     */
    it('has a className property with value ReactVirtualized__Collection', () => {
      expect(wrapper.find('.ReactVirtualized__Collection')).toHaveLength(1);
      wrapper.unmount();
    });
  });
  it('should select, focus, and scroll to given item index', () => {
    mockGetWidth.mockReturnValueOnce(660);
    const wrapper = mount(
      <ElementList {...props} selectedItemIndex={5} focusedItemIndex={5} />,
    );
    // @ts-ignore TS2339: scrollToCell prop exists on Collection component
    expect(wrapper.find('Collection').props().scrollToCell).toBe(5);
  });
});
