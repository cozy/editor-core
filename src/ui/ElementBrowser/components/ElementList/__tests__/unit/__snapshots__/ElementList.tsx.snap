// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElementList should match snapshot 1`] = `
.c0 {
  position: relative;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  overflow: hidden;
  padding: 2px;
}

.c1 .ReactVirtualized__Collection {
  -webkit-scrollbar-color: #eeeeee rgba(255,255,255,0);
  -moz-scrollbar-color: #eeeeee rgba(255,255,255,0);
  -ms-scrollbar-color: #eeeeee rgba(255,255,255,0);
  scrollbar-color: #eeeeee rgba(255,255,255,0);
  -ms-overflow-style: -ms-autohiding-scrollbar;
  border-radius: 3px;
  outline: none;
}

.c1 .ReactVirtualized__Collection::-webkit-scrollbar {
  width: 8px;
}

.c1 .ReactVirtualized__Collection::-webkit-scrollbar-track-piece {
  background: rgba(255,255,255,0);
}

.c1 .ReactVirtualized__Collection::-webkit-scrollbar-thumb {
  background: #eeeeee;
}

.c1 .ReactVirtualized__Collection:focus {
  box-shadow: 0 0 0 2px #4C9AFF;
}

.c1 .ReactVirtualized__Collection__innerScrollContainer div[class='element-item-wrapper']:last-child {
  padding-bottom: 4px;
}

<WithAnalyticsContext(ElementList)
  items={
    Array [
      Object {
        "action": [MockFunction],
        "name": "item-1",
        "title": "Item 1",
      },
      Object {
        "action": [MockFunction],
        "name": "item-2",
        "title": "Item 2",
      },
      Object {
        "action": [MockFunction],
        "name": "item-3",
        "title": "Item 3",
      },
      Object {
        "action": [MockFunction],
        "name": "item-4",
        "title": "Item 4",
      },
      Object {
        "action": [MockFunction],
        "name": "item-5",
        "title": "Item 5",
      },
    ]
  }
  mode="full"
  onEnterKeyPress={[MockFunction]}
  onInsertItem={[MockFunction]}
  selectedItemIndex={1}
  setColumnCount={[MockFunction]}
  setFocusedItemIndex={[MockFunction]}
  setSelectedItemIndex={[MockFunction]}
>
  <AnalyticsContext
    data={
      Object {
        "component": "ElementList",
      }
    }
  >
    <ElementList
      items={
        Array [
          Object {
            "action": [MockFunction],
            "name": "item-1",
            "title": "Item 1",
          },
          Object {
            "action": [MockFunction],
            "name": "item-2",
            "title": "Item 2",
          },
          Object {
            "action": [MockFunction],
            "name": "item-3",
            "title": "Item 3",
          },
          Object {
            "action": [MockFunction],
            "name": "item-4",
            "title": "Item 4",
          },
          Object {
            "action": [MockFunction],
            "name": "item-5",
            "title": "Item 5",
          },
        ]
      }
      mode="full"
      onEnterKeyPress={[MockFunction]}
      onInsertItem={[MockFunction]}
      selectedItemIndex={1}
      setColumnCount={[MockFunction]}
      setFocusedItemIndex={[MockFunction]}
      setSelectedItemIndex={[MockFunction]}
    >
      <Memo()>
        <WidthObserverWrapper
          innerRef={
            Object {
              "current": <div
                class="c0"
                tabindex="-1"
              >
                <div />
              </div>,
            }
          }
          tabIndex={-1}
        >
          <div
            className="c0"
            tabIndex={-1}
          >
            <Memo()
              setWidth={[Function]}
            >
              <Memo()
                setWidth={[Function]}
                useIntersectionObserver={false}
              >
                <Memo()
                  setWidth={[Function]}
                  subscribe={[Function]}
                >
                  <div />
                </Memo()>
              </Memo()>
            </Memo()>
          </div>
        </WidthObserverWrapper>
      </Memo()>
      <styled.div
        data-testid="element-items"
      >
        <div
          className="c1"
          data-testid="element-items"
        >
          <ThemeProvider
            theme={
              Object {
                "@atlaskit-shared-theme/item": Object {
                  "-ms-flex": "auto",
                  "beforeItemSpacing": Object {
                    "default": 12,
                  },
                  "borderRadius": 4,
                  "boxSizing": "border-box",
                  "height": Object {
                    "default": 75,
                  },
                  "hover": Object {
                    "background": "rgb(244, 245, 247)",
                  },
                  "padding": Object {
                    "default": Object {
                      "bottom": 12,
                      "left": 12,
                      "right": 12,
                      "top": 12,
                    },
                  },
                  "position": "relative",
                  "selected": Object {
                    "background": "#F4F5F7",
                  },
                },
              }
            }
          />
        </div>
      </styled.div>
    </ElementList>
  </AnalyticsContext>
</WithAnalyticsContext(ElementList)>
`;
